tests/components/homekit_controller/test_device_trigger.py

```diff
@@ -0,0 +1,298 @@
+                }
+            )
+
+    triggers = await async_get_device_automations(hass, "trigger", device.id)
+    assert_lists_same(triggers, expected)
+
+
+async def test_enumerate_button(hass, utcnow):
+    """Test that a button is correctly enumerated."""
+    await setup_test_component(hass, create_button)
+
+    entity_registry = await hass.helpers.entity_registry.async_get_registry()
+    entry = entity_registry.async_get("sensor.testdevice_battery")
+
+    device_registry = await hass.helpers.device_registry.async_get_registry()
+    device = device_registry.async_get(entry.device_id)
+
+    expected = [
+        {
+            "device_id": device.id,
+            "domain": "sensor",
+            "entity_id": "sensor.testdevice_battery",
+            "platform": "device",
+            "type": "battery_level",
+        }
+    ]
+
+    for subtype in ("single_press", "double_press", "long_press"):
+        expected.append(
+            {
+                "device_id": device.id,
+                "domain": "homekit_controller",
+                "platform": "device",
+                "type": "button1",
+                "subtype": subtype,
+            }
+        )
+
+    triggers = await async_get_device_automations(hass, "trigger", device.id)
+    assert_lists_same(triggers, expected)
+
+
+async def test_enumerate_doorbell(hass, utcnow):
+    """Test that a button is correctly enumerated."""
+    await setup_test_component(hass, create_doorbell)
+
+    entity_registry = await hass.helpers.entity_registry.async_get_registry()
+    entry = entity_registry.async_get("sensor.testdevice_battery")
+
+    device_registry = await hass.helpers.device_registry.async_get_registry()
+    device = device_registry.async_get(entry.device_id)
+
+    expected = [
+        {
+            "device_id": device.id,
+            "domain": "sensor",
+            "entity_id": "sensor.testdevice_battery",
+            "platform": "device",
+            "type": "battery_level",
+        }
+    ]
+
+    for subtype in ("single_press", "double_press", "long_press"):
+        expected.append(
+            {
+                "device_id": device.id,
+                "domain": "homekit_controller",
+                "platform": "device",
+                "type": "doorbell",
+                "subtype": subtype,
+            }
+        )
+
+    triggers = await async_get_device_automations(hass, "trigger", device.id)
+    assert_lists_same(triggers, expected)
+
+
+async def test_handle_events(hass, utcnow, calls):
+    """Test that events are handled."""
+    helper = await setup_test_component(hass, create_remote)
+
+    entity_registry = await hass.helpers.entity_registry.async_get_registry()
+    entry = entity_registry.async_get("sensor.testdevice_battery")
+
+    device_registry = await hass.helpers.device_registry.async_get_registry()
+    device = device_registry.async_get(entry.device_id)
+
+    assert await async_setup_component(
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6158285)